api/src/main/java/jakarta/data/repository/BasicRepository.java

```diff
@@ -146,6 +146,15 @@ public interface BasicRepository<T, K> extends DataRepository<T, K> {
     @Find
     Optional<T> findById(@By(ID) K id);
 
+    /**
+     * Retrieves the entities with the given unique identifiers.
+     *
+     * @param ids a list of unique identifiers.
+     * @return the entities with the given unique identifiers.
+     */
+    @Query("where id(this) in :ids order by id(this)")
+    List<T> findByIdIn(@By(ID) List<K> ids);
```

Comment:
Why put this method back again?
We agree to remove it.

Reply:
Well, we decided to remove it because we didn't have a good way to express it in any of the available query facilities, not because we hated it and thought it was useless.

I think it's kinda, well, at least medium-useful. Not a _critical_ thing to have, but not a wart either.